### PR TITLE
Update 2.3 system requirements

### PR DESCRIPTION
**Pull request by jeff-matthews**
## Purpose of this pull request

This pull request (PR) updates 2.3 system requirements similar to #7529. 

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.3/install-gde/system-requirements-tech.html

whatsnew
Updated Magento [system requirements](https://devdocs.magento.com/guides/v2.3/install-gde/system-requirements-tech.html) for MySQL, PHP, and Redis.

### CODE REVIEW

File: src/guides/v2.3/install-gde/system-requirements-tech.md [hunk 1]
```diff
@@ -36,23 +36,22 @@ Upgrading the Magento applications and extensions you obtain from Magento Market
 
 ## Database
 
-MySQL 5.6, 5.7
+*  MySQL 5.7 or on-premise installations
```
Comment (meker12) [inline src/guides/v2.3/install-gde/system-requirements-tech.md:39]:
```suggestion
*  MySQL 5.7 for on-premise installations
```

File: src/guides/v2.3/install-gde/system-requirements-tech.md [hunk 2]
```diff
@@ -36,23 +36,22 @@ Upgrading the Magento applications and extensions you obtain from Magento Market
...
-Magento is also compatible with MySQL NDB Cluster 7.4.*, MariaDB 10.0, 10.1, 10.2, Percona 5.7, and other binary-compatible MySQL technologies.
+Magento is also compatible with, but has not been tested and is not recommended, with MySQL 5.6, MariaDB 10.1, and Percona 5.7.
 
 {:.bs-callout-info}
 Magento only uses MySQL features compatible with MariaDB. MariaDB may not be compatible with all MySQL features, however, so be sure to research compatibility issues before using a feature in your Magento module.
 
 ## PHP
 
+*  PHP ~7.3.0
+
```
Comment (meker12) [inline src/guides/v2.3/install-gde/system-requirements-tech.md:50]: what about 7.4 (we have it on the Cloud versions table, but wanted to check to see if we have a mistake.)

File: src/guides/v2.3/install-gde/system-requirements-tech.md [hunk 3]
```diff
@@ -36,23 +36,22 @@ Upgrading the Magento applications and extensions you obtain from Magento Market
 
 ## Database
 
-MySQL 5.6, 5.7
+*  MySQL 5.7 or on-premise installations
+*  MariaDB 10.2 for {{site.data.var.ece}} projects
```
Comment (meker12) [inline src/guides/v2.3/install-gde/system-requirements-tech.md:40]: For Cloud, the versions table shows MariaDB 10.2, 10.3, and 10.4

File: src/guides/v2.3/install-gde/system-requirements-tech.md (hunk 2 above)
Comment (jeff-matthews) [reply src/guides/v2.3/install-gde/system-requirements-tech.md:50]: Product Management requested that we leave PHP ~7.3.0 only for Magento 2.3.x.

File: src/guides/v2.3/install-gde/system-requirements-tech.md (hunk 3 above)
Comment (jeff-matthews) [reply src/guides/v2.3/install-gde/system-requirements-tech.md:40]: Product Management requested this change

Comment (meker12) [reply src/guides/v2.3/install-gde/system-requirements-tech.md:40]: Ignore me-- 10.3 and 10.4 are for 2.4.

File: src/guides/v2.3/install-gde/system-requirements-tech.md (hunk 2 above)
Comment (meker12) [reply src/guides/v2.3/install-gde/system-requirements-tech.md:50]: 7.4 is for 2.4 -- never mind.

File: src/guides/v2.3/install-gde/system-requirements-tech.md [hunk 4]
```diff
@@ -36,23 +36,22 @@ Upgrading the Magento applications and extensions you obtain from Magento Market
 
 ## Database
 
-MySQL 5.6, 5.7
+*  MySQL 5.7 for on-premise installations
```
Comment (hguthrie) [inline src/guides/v2.3/install-gde/system-requirements-tech.md:39]: shouldn't this be "on-premises" ? 😇

Comment (jeff-matthews) [reply src/guides/v2.3/install-gde/system-requirements-tech.md:39]: I think [this post](https://www.adamfowlerit.com/2017/04/premise-vs-premises-cares/#:~:text=Yes%2C%20the%20word%20%E2%80%9Cpremises%E2%80%9D,considered%20in%20an%20official%20context.&text=so%2C%20it%20makes%20sense%20to,the%20correct%20term%20to%20use.) sums up my feelings about that.

Adobe uses "on-premise" as well, so I'm inclined to stick with that: https://helpx.adobe.com/legal/product-descriptions/adobe-experience-manager-on-premise.html.

Comment (meker12) [reply src/guides/v2.3/install-gde/system-requirements-tech.md:39]: This did come up, and I researched it. "On-premises is in fact correct.  See https://collectivecontent.co.uk/2018/04/19/should-i-say-on-premise-or-on-premises-it/  

 Google search results:  
on-premises: 205,000,000
on-premise 86,900,000 

Thanks for remembering @hguthrie.  Also, it's not a show-stopper, but it's nice to do things correctly when we can.

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**